gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_executorInstance/LocalFsSpecProducer.java

```diff
@@ -0,0 +1,106 @@
+import java.util.Properties;
+import java.util.concurrent.Future;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.gobblin.runtime.api.JobSpec;
+import org.apache.gobblin.runtime.api.Spec;
+import org.apache.gobblin.runtime.api.SpecExecutor;
+import org.apache.gobblin.runtime.api.SpecProducer;
+import org.apache.gobblin.util.CompletedFuture;
+import org.apache.gobblin.util.ConfigUtils;
+
+
+/**
+ * An implementation of {@link SpecProducer} that produces {@link JobSpec}s to the {@value #LOCAL_FS_PRODUCER_PATH_KEY}
+ */
+@Slf4j
+public class LocalFsSpecProducer implements SpecProducer<Spec> {
+  private String specProducerPath;
+  public static final String LOCAL_FS_PRODUCER_PATH_KEY = "gobblin.cluster.localSpecProducer.dir";
+
+  public LocalFsSpecProducer(Config config) {
+    this.specProducerPath = ConfigUtils.getString(config, LOCAL_FS_PRODUCER_PATH_KEY, "");
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(this.specProducerPath), "Missing argument: " + LOCAL_FS_PRODUCER_PATH_KEY);
+    File parentDir = new File(specProducerPath);
+    if (!parentDir.exists()) {
+      if (parentDir.mkdirs()) {
+        log.info("Creating directory path at {}", this.specProducerPath);
+      }
+    }
+  }
+
+  /** Add a {@link Spec} for execution on {@link org.apache.gobblin.runtime.api.SpecExecutor}.
+   * @param addedSpec*/
+  @Override
+  public Future<?> addSpec(Spec addedSpec) {
+    return writeSpec(addedSpec, SpecExecutor.Verb.ADD);
+  }
+
+  /** Update a {@link Spec} being executed on {@link org.apache.gobblin.runtime.api.SpecExecutor}.
+   * @param updatedSpec*/
+  @Override
+  public Future<?> updateSpec(Spec updatedSpec) {
+    return writeSpec(updatedSpec, SpecExecutor.Verb.UPDATE);
+  }
+
+  private Future<?> writeSpec(Spec spec, SpecExecutor.Verb verb) {
+    if (spec instanceof JobSpec) {
+      String specString = spec.toString();
+      // format the JobSpec to have file of <flowGroup>_<flowName>.job
+      String jobFileName = specString.replace('/', '_').substring(specString.lastIndexOf(':')+2, specString.length()-1) + ".job";
+      try (
+        FileOutputStream fStream = new FileOutputStream(this.specProducerPath + File.separatorChar + jobFileName);
+      ) {
+        ((JobSpec) spec).getConfigAsProperties().store(fStream, null);
+        log.info("Writing job {} to {}", jobFileName, this.specProducerPath);
+        return new CompletedFuture<>(Boolean.TRUE, null);
+      } catch (IOException e) {
+        log.error("Exception encountered when adding Spec {}", spec);
+        return new CompletedFuture<>(Boolean.TRUE, e);
+      }
+    } else {
+      throw new RuntimeException("Unsupported spec type " + spec.getClass());
+    }
+  }
+
+  /** Delete a {@link Spec} being executed on {@link org.apache.gobblin.runtime.api.SpecExecutor}.
+   * @param deletedSpecURI
+   * @param headers*/
+  @Override
+  public Future<?> deleteSpec(URI deletedSpecURI, Properties headers) {
+    return new CompletedFuture<>(Boolean.TRUE, null);
```

Comment:
are we expected to delete the job corresponds to the uri?

Reply:
Ah oops I should implement that, good point